File: Iter2/src/Course.java
import java.util.ArrayList;

public abstract class Course {
	
    // course id
    public String courseId;
    
     // Schedule of course
     
    public ArrayList<int[]> courseSchedule = new ArrayList<>();
    
   //hold registered students in an array
    public ArrayList<Student> registeredStudents = new ArrayList<>();

    //get course id of the course
    public String getCourseId() {
        return courseId;
    }

   //get course schedule of the course
    public ArrayList<int[]> getCourseSchedule() {
        return courseSchedule;
    }

    
     //Create the course schedule in days and hours
     
    public void addLectureHour(int day, int lectureHour) {
        courseSchedule.add(new int[]{day, lectureHour});
    }

    
     //Add a new student to the course enrollment list
    public void addStudent(Student student) {
        registeredStudents.add(student);
    }
}

File: Iter3/src/course.py
from abc import ABC

class Course(ABC):
    courseID: str
    courseSchedule = []
    __registeredStudents = []

    def get_course_id(self):
        return self.courseID

    def get_course_schedule(self):
        return self.courseSchedule

    def add_student(self, student):
        self.__registeredStudents.append(student)

    def add_lecture_hour(self, day: int, lecture_hour: int):
        self.courseSchedule.append({day, lecture_hour})

